### PR TITLE
Bump npm dependencies (npm test passed!)

### PR DESCRIPTION
**Pull request by PeterDaveHello**

### CODE REVIEW

**Comment by thlorenz** [comment]
Please use `~` ranges, otherwise I can't accept this.

**Comment by PeterDaveHello** [comment]
@thlorenz is there any concern about using `^`? Thanks.

**Comment by thlorenz** [comment]
Yes, opens up to more breakage as people don't perfectly follow semver. I prefer to be more conservative and therefore would like to stick to `~`.
If you don't agree, fine, but I won't merge this since I don't wanna deal with lots of breakage issues in the future.

Thanks.

**Comment by PeterDaveHello** [comment]
So do we need to change the other dependencies also to use `~`? Thanks.

**Comment by thlorenz** [comment]
Ideally yes, wasn't aware that some of them had a `^` at this point.
Most likely got introduced with another PR and I missed it :)

**Comment by PeterDaveHello** [comment]
I add a commit to use `~` in all the dependencies now

File: package.json [hunk 1]
```diff
@@ -16,15 +16,15 @@
...
-    "htmlparser2": "~3.7.1",
-    "markdown-to-ast": "~3.2.3",
-    "minimist": "~1.1.0",
-    "underscore": ">=1.3.3",
-    "update-section": "^0.3.0"
+    "anchor-markdown-header": "~0.5.5",
+    "htmlparser2": "~3.9.2",
+    "markdown-to-ast": "~3.4.0",
+    "minimist": "~1.2.0",
+    "underscore": "~1.3.3",
```
Comment (thlorenz) [inline package.json]: what's the latest underscore? We need to make sure we don't downgrade the version here.
Should be `~<whatever latest version is>`.

File: package.json [hunk 2]
```diff
@@ -16,15 +16,15 @@
     "node": ">=0.4"
   },
   "dependencies": {
-    "anchor-markdown-header": "^0.5.5",
-    "htmlparser2": "~3.7.1",
-    "markdown-to-ast": "~3.2.3",
-    "minimist": "~1.1.0",
-    "underscore": ">=1.3.3",
-    "update-section": "^0.3.0"
+    "anchor-markdown-header": "~0.5.5",
```
Comment (thlorenz) [inline package.json]: My repo, so we can use `^` here (I trust myself to follow semver ;) )

File: package.json [hunk 3]
```diff
@@ -16,15 +16,15 @@
...
+    "htmlparser2": "~3.9.2",
+    "markdown-to-ast": "~3.4.0",
+    "minimist": "~1.2.0",
+    "underscore": "~1.3.3",
+    "update-section": "~0.3.0"
```
Comment (thlorenz) [inline package.json]: My repo, so we can use `^` here (I trust myself to follow semver ;) )

Comment (PeterDaveHello) [inline package.json]: updated

**Comment by thlorenz** [comment]
Awesome, many thanks :)
Please squash into one commit first and then merge to master.

I made you collaborator.
So when you merge please follow [this guide](https://gist.github.com/thlorenz/f7df0daf13c6f6c1fa84).
I will then version and publish to npm.

Thanks.

**Comment by PeterDaveHello** [comment]
Hi @thlorenz, IMO, these two commits are doing different their own task, one for version bump and one for syntax, can we just leave them be separated? Thanks.

**Comment by thlorenz** [comment]
@PeterDaveHello yeah, makes sense. Sounds good to me, go ahead and merge then.

**Comment by PeterDaveHello** [comment]
@thlorenz I'm confused about how should I merge it since the reference doc says not to use the green button, this PR branch is up to date, so can I merge it right here? Thanks.

**Comment by thlorenz** [comment]
Merge using `git` on the terminal, i.e. you'd do:

```
git rebase master
git co master
git merge dependencies-upgrade
```

It's guaranteed to be fast forward merge since you are rebasing first and if you have any conflicts you get to fix them right there.
I'm mentioning alternative ways to fast forward merges, including some links to more details [here](https://gist.github.com/thlorenz/f7df0daf13c6f6c1fa84#merging-a-pr)

**Comment by thlorenz** [comment]
You could also just hit the `Rebase and merge` option in the button dropdown, but learning to do this on the terminal is a good thing to do in general.

**Comment by PeterDaveHello** [comment]
@thlorenz Actually I totally have no problem on how to merging locally, just want to make sure the policy and procedure, if I merge it locally, will this PR also shown as "Merged"? Or it or just be closed? Thanks.

**Comment by thlorenz** [comment]
Thanks went out as 1.3 .. I did a minor upgrade since we changed a gzillion deps.

**Pull request merged**